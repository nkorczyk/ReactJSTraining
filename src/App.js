import React, { Component } from 'react';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>Movies</h1>
        <p>List of movies</p>
      </div>
    );
  }
}

export default App;
