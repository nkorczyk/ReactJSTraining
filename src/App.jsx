import React, { Component } from 'react';
import './style.css';
import 'materialize-css';
import 'materialize-css/dist/css/materialize.min.css';
import Header from './components/Header';
import Results from './components/Results';
import Content from './components/Content';
import Footer from './components/Footer';
import ErrorBoundary from './components/ErrorBoundary';
import axios from 'axios';

class App extends Component {
  state = {
    data: []
  };

  componentDidMount() {
    axios.get('http://react-cdp-api.herokuapp.com/movies')
      .then(response => {
        this.setState({
          data: response.data.data.slice(0, 9)
        });
      });
  }

  render() {
    return (
      <React.Fragment>
        <Header />
        <Results />
        <ErrorBoundary>
          <Content movies={this.state.data} />
        </ErrorBoundary>
        <Footer />
      </React.Fragment>
    );
  }
}

export default App;
