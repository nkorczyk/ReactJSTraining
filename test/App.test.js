import React from 'react';
import TestRenderer from 'react-test-renderer';
import { mount } from 'enzyme';
import App from "../src/App";

describe('App Snapshot', () => {
  test('renders', () => {
    const component = TestRenderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
