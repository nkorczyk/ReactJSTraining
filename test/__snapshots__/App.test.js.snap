// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot renders 1`] = `
<div>
  <header
    className="header"
  >
    <div>
      <h5
        className="red-text text-darken-2"
      >
        netflixroulette
      </h5>
    </div>
    <h4
      className="white-text text-darken-2"
    >
      FIND YOUR MOVIE
    </h4>
    <input
      className="input-field white-text"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Enter the title"
      type="text"
    />
    <div>
      <h6
        className="white-text text-darken-2 left"
      >
        SEARCH BY
      </h6>
      <button
        className="lighten-1 btn buttons red"
        id="title"
        onClick={[Function]}
      >
        TITLE
      </button>
      <button
        className="lighten-1 btn buttons grey"
        id="genre"
        onClick={[Function]}
      >
        GENRE
      </button>
      <button
        className="red lighten-1 btn right"
        id="search"
        onClick={[Function]}
      >
        SEARCH
      </button>
    </div>
  </header>
  <div
    className="bar grey lighten-1"
  >
    <h6
      className="white-text text-darken-2 grey lighten-1 filter-bar left"
    >
      0 movies found
    </h6>
    <h6
      className="white-text text-darken-2 grey lighten-1 filter-bar right"
    >
      rating
    </h6>
    <h6
      className="white-text text-darken-2 grey lighten-1 filter-bar right"
    >
      release date
    </h6>
    <h6
      className="white-text text-darken-2 grey lighten-1 filter-bar right"
    >
      Sort by
    </h6>
  </div>
  <div
    className="not-found"
  >
    <h3
      className="not-found-message"
    >
      No Films found
    </h3>
  </div>
  <footer
    className="bar"
  >
    <h5
      className="red-text text-darken-2 grey darken-3 grey-bar"
    >
      netflixroulette
    </h5>
  </footer>
</div>
`;
